src/transformState.js

```diff
@@ -62,7 +62,24 @@
  * @param {Object[]} transforms
  */
 function transformState(state, transforms) {
-  // write code here
+  for (const transform of transforms) {
+    switch (transform['operation']) {
+      case 'addProperties':
+        for (const key in transform['properties']) {
+          state[key] = transform['properties'][key];
+        }
+        break;
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+      case 'removeProperties':
+        for (const key in transform['properties']) {
```

Comment:
better to create a variable for transform['properties'].
state[transform['properties'][key]] is hard to read